extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/LiquibaseFactory.java

```diff
@@ -31,9 +34,23 @@ public Liquibase createLiquibase() {
         try (ClassLoaderResourceAccessor resourceAccessor = new ClassLoaderResourceAccessor(
                 Thread.currentThread().getContextClassLoader())) {
 
-            Database database = DatabaseFactory.getInstance()
-                    .findCorrectDatabaseImplementation(new JdbcConnection(dataSource.getConnection()));
-            ;
+            Database database;
+
+            if (config.username.isPresent() && config.password.isPresent()) {
+                AgroalDataSource agroalDataSource = (AgroalDataSource) dataSource;
+                String jdbcUrl = agroalDataSource.getConfiguration().connectionPoolConfiguration()
+                        .connectionFactoryConfiguration().jdbcUrl();
+                Connection connection = DriverManager.getConnection(jdbcUrl, config.username.get(), config.password.get());
```

Comment:
Can we use `datasource.getConnection(String username, String password)` here instead of using the Agroal API?

Reply:
No Than the error, in this https://github.com/quarkusio/quarkus/issues/31214#issuecomment-1991117911  discussion occurs.